react/features/whiteboard/index.tsx

```diff
@@ -0,0 +1,144 @@
+// @ts-ignore
+import { getLocalParticipant } from '../base/participants';
+import { USERNAME_UPDATE_STATUS } from './constants';
+
+const HEIGHT_OFFSET = 80;
+
+declare const interfaceConfig: any;
+
+interface IDimensions {
+
+    /* The height of the component. */
+    height: string;
+
+    /* The width of the component. */
+    width: string;
+}
+
+/**
+ * The Whiteboard component.
+ *
+ * @returns {JSX.Element} - The React component.
+ */
+const Whiteboard: () => JSX.Element = () => {
+    const excalidrawRef = useRef<any>(null);
+    const collabAPIRef = useRef<any>(null);
+    const dispatch = useDispatch();
+
+    const shouldDisplay = useSelector(isWhiteboardVisible);
+    const isInTileView = useSelector(shouldDisplayTileView);
+    const { clientHeight, clientWidth } = useSelector((state: any) => state['features/base/responsive-ui']);
+    const { visible: filmstripVisible, isResizing } = useSelector((state: any) => state['features/filmstrip']);
+    const filmstripWidth: number = useSelector(getVerticalViewMaxWidth);
+    const collabLink = useSelector(getCollabLink);
+    const collabServerUrl = useSelector(getCollabServerUrl);
+    // @ts-ignore
+    const localParticipantName: string = useSelector(getLocalParticipant).name;
+    const usernameStatus = useSelector(getUsernameStatus);
+
+    useEffect(() => {
+        if (!collabAPIRef.current || usernameStatus !== 'NEW') {
+            return;
+        }
+
+        collabAPIRef.current.setUsername(localParticipantName);
+        dispatch(setUsernameStatus(USERNAME_UPDATE_STATUS.UPDATED));
+    }, [ usernameStatus ]);
+
+    /**
+    * Computes the width and the height of the component.
+    *
+    * @returns {IDimensions} - The dimensions of the component.
+    */
+    const getDimensions = (): IDimensions => {
+        let width: number;
+        let height: number;
+
+        if (interfaceConfig.VERTICAL_FILMSTRIP) {
+            if (filmstripVisible) {
+                width = clientWidth - filmstripWidth;
+            } else {
+                width = clientWidth;
+            }
+            height = clientHeight - getToolboxHeight();
+        } else {
+            if (filmstripVisible) {
+                height = clientHeight - Filmstrip.getFilmstripHeight();
+            } else {
+                height = clientHeight;
+            }
+            width = clientWidth;
+        }
+
+        return {
+            width: `${width}px`,
+            height: `${height - HEIGHT_OFFSET}px`
+        };
+    };
+
+    const getCollabAPI = useCallback(collabAPI => {
+        if (collabAPIRef.current) {
+            return;
+        }
+        collabAPIRef.current = collabAPI;
+    }, []);
+
+    return (
+        <div
+            className = { clsx(isResizing && 'disable-pointer', 'whiteboard-container') }
+            style = {{
+                ...getDimensions(),
+                marginTop: `${HEIGHT_OFFSET}px`,
+                display: `${isInTileView ? 'none' : 'block'}`
```

Comment:
I wonder if we should display it in tile view.

Reply:
We discussed that we'll display only an icon for now in tile view.

Reply:
👍